pennylane/transforms/op_transforms.py

```diff
@@ -0,0 +1,481 @@
+
+        .. code-block:: python
+
+            @qml.op_transform
+            def name(op, lower=False):
+                if lower:
+                    return op.name.lower()
+                return op.name
+
+            @name.tape_transform
+            def name(tape, lower=True):
+                return [name(op, lower=lower) for op in tape.operations]
+
+        We can now use this function on a qfunc, tape, or QNode:
+
+        >>> def circuit(x, y):
+        ...     qml.RX(x, wires=0)
+        ...     qml.Hadamard(wires=1)
+        ...     qml.CNOT(wires=[0, 1])
+        ...     qml.CRY(y, wires=[1, 0])
+        >>> name(circuit, lower=True)(0.1, 0.8)
+        ['rx', 'hadamard', 'cnot', 'cry']
+
+        If the transformation has purely quantum output, we can register the tape transformation
+        as a qfunc transformation in addition:
+
+        .. code-block:: python
+
+            @qml.op_transform
+            def simplify_rotation(op):
+                if op.name == "Rot":
+                    params = op.parameters
+                    wires = op.wires
+
+                    if qml.math.allclose(params, 0):
+                        return
+
+                    if qml.math.allclose(params[1:2], 0):
+                        return qml.RZ(params[0], wires)
+
+                return op
+
+            @simplify_rotation.tape_transform
+            @qml.qfunc_transform
+            def simplify_rotation(tape):
+                for op in tape.operations + tape.measurements:
+                    if op.name == "Rot":
+                        simplify_rotation(op)
+                    else:
+                        qml.apply(op)
+
+        We can now use this combined operator and quantum function transform in compilation pipelines:
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            @qml.compile(pipeline=[simplify_rotation])
+            def circuit(weights):
+                ansatz(weights)
+                qml.CNOT(wires=[0, 1])
+                qml.Rot(0.0, 0.0, 0.0, wires=0)
+                return qml.expval(qml.PauliX(1))
+        """
+        self._tape_fn = fn
+        return self
+
+    def _create_wrapper(self, obj, *targs, wire_order=None, **tkwargs):
+
+        if isinstance(obj, qml.operation.Operator):
+            # Input is a single operation.
+            # op_transform(obj, *transform_args)
+            if wire_order is not None:
+                tkwargs["wire_order"] = wire_order
+
+            wrapper = self.fn(obj, *targs, **tkwargs)
+
+        elif isinstance(obj, qml.tape.QuantumTape):
+            # Input is a quantum tape. Get the quantum tape.
+            tape, verified_wire_order = _make_tape(obj, wire_order)
+
+            if wire_order is not None:
+                tkwargs["wire_order"] = verified_wire_order
+
+            wrapper = self.tape_fn(tape, *targs, **tkwargs)
+
+        elif callable(obj):
+            # Input is a QNode, or qfunc. Get the quantum tape.

```

Comment:
Could it not be a subclass of `Operator`?

Reply:
Yep, a subclass of `Operator` _is_ a qfunc, albeit a qfunc that only contains a single operation!

Reply:
Got it! Maybe worth noting that here.

Reply:
```suggestion
            # Input is a QNode, or qfunc (including single-operation qfuncs).
            # Get the quantum tape.
```